packages/headless/src/features/debug/ranking-info-parser.ts

```diff
@@ -1,44 +1,160 @@
-export type RankingInfo = ReturnType<typeof parseRankingInfo>;
+export interface RankingInformation {
+  /**
+   * The attributes of the document that contributed to its ranking.
+   * */
+  documentWeights: DocumentWeights | null;
 
-export interface ListOfWeights {
+  /**
+   * The weight attributed to each term in the query.
+   */
+  termsWeight: TermWeightReport | null;
+
+  /**
+   * The sum of all weights.
+   */
+  totalWeight: number | null;
+
+  /**
+   * The weights applied by query ranking expressions.
+   */
+  qreWeights: QueryRankingExpressionWeights[];
+}
+
+export interface DocumentWeights {
+  /**
+   * The effect of proximity of query terms. More weight is given to documents having the terms closer together.
+   */
   Adjacency: number;
+
+  /**
+   * The weight attributed to user ratings.
+   */
   'Collaborative rating': number;
+
+  /**
+   * The weight assigned through an [indexing pipeline extension (IPE)](https://docs.coveo.com/en/206/) for the item.
+   */
   Custom: number;
+
+  /**
+   * The weight attributed to the document date.
+   */
   Date: number;
+
+  /**
+   * The weight applied by a [query ranking expression (QRE)](https://docs.coveo.com/en/2777/coveo-solutions/using-query-ranking-expressions).
+   */
   QRE: number;
+
+  /**
+   * The proximity of the document in the remaining results after filtering indexed items by query terms and user permissions.
```

Comment:
I don't understand what 'proximity' means here. Proximity to what?

Reply:
My understanding is after the first filtration step that collects the top 50,000 documents, Quality captures how close an particular document is to the root/first document.

I took the word `proximity` from the table on the [this page](https://docs.coveo.com/en/1624/searching-with-coveo/understanding-search-result-ranking#pre-tuned-ranking-weight-factors). I cannot link to the exact row, but it is under Phase 2 -> Item Quality Evaluation.

I think I will change `proximity` to `rank` for now,